doc/threading_model.md

```diff
@@ -0,0 +1,31 @@
+﻿# gRPC Threading Model

```

Comment:
This title should also say explicitly gRPC C++ Threading model with a filename of cpp_threading_model.md as the rules are somewhat different in other APIs.